test/integration/scheduler/podsready/scheduler_test.go

```diff
@@ -204,14 +204,17 @@ var _ = ginkgo.Describe("SchedulerWithWaitForPodsReady", func() {
 			ginkgo.By("determining the time of admission as LastTransitionTime for the Admitted condition")
 			admittedAt := apimeta.FindStatusCondition(prodWl1.Status.Conditions, kueue.WorkloadAdmitted).LastTransitionTime.Time
 
-			ginkgo.By("wait for the 'prod1' workload to be unadmitted")
-			gomega.Eventually(func() *kueue.Admission {
+			ginkgo.By("wait for the 'prod1' workload to be evicted")
+			gomega.Eventually(func() bool {
 				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(prodWl1), prodWl1)).Should(gomega.Succeed())
+				isEvicting := apimeta.IsStatusConditionTrue(prodWl1.Status.Conditions, kueue.WorkloadEvicted)
 				if time.Since(admittedAt) < podsReadyTimeout {
-					gomega.Expect(prodWl1.Status.Admission).ShouldNot(gomega.BeNil())
+					gomega.Expect(isEvicting).Should(gomega.BeFalse())
 				}
-				return prodWl1.Status.Admission
-			}, util.Timeout, util.Interval).Should(gomega.BeNil())
+				return isEvicting
+			}, util.Timeout, util.Interval).Should(gomega.BeTrue())
```

Comment:
Avoid comparing a boolean, as this gives very little context in error messages.

you could, for example, return the condition itself and use `gomega.BeComparableTo`.

Otherwise, at least add an accompanying error message to `Should`.